usaspending_api/search/filters/postgres/tas.py

```diff
@@ -0,0 +1,118 @@
+
+
+def string_to_dictionary(string, true_agency_name):
+    if len(string.split("-")) == 1:
+        return {true_agency_name: string}
+    elif len(string.split("-")) == 2:
+        return FederalAccount.fa_rendering_label_to_component_dictionary(string)
+    else:
+        return TreasuryAppropriationAccount.tas_rendering_label_to_component_dictionary(string)
+
+
+def search_regex_of(v):
+    if isinstance(v, str):
+        v = string_to_dictionary(v, "aid")
+
+    code_lookup = {
+        "ata": v["ata"] if v.get("ata") else None,
+        "aid": v["aid"] if v.get("aid") else ".*",
+        "main": v["main"] if v.get("main") else ".*",
+        "sub": v["sub"] if v.get("sub") else ".*",
+        "bpoa": v["bpoa"] if v.get("bpoa") else ".*",
+        "epoa": v["epoa"] if v.get("epoa") else ".*",
+        "a": v["a"] if v.get("a") else ".*",
+    }
+
+    # This is NOT the order of elements as displayed in the tas rendering label, but instead the order in the award_delta_view and transaction_delta_view
+    search_regex = TreasuryAppropriationAccount.generate_tas_rendering_label(
+        code_lookup["ata"],
+        code_lookup["aid"],
+        code_lookup["a"],
+        code_lookup["bpoa"],
+        code_lookup["epoa"],
+        code_lookup["main"],
+        code_lookup["sub"],
+    )
+
+    # TODO: move this to a Tinyshield filter
+    if not re.match(r"^(\d|\w|-|\*|\.)+$", search_regex):
+        raise UnprocessableEntityException(f"Unable to parse TAS filter {search_regex}")
+
+    return search_regex
+
+
+class TASNode(Node):
+    def _basic_search_unit(self):
+        if len(self.ancestors) == 2:
+            return Q(**{"tas_rendering_label__iregex": search_regex_of(self.code)})
+        elif len(self.ancestors) == 1:
+            return Q(**{"federal_account__federal_account_code": self.code})
+        else:
+            return Q(**{"federal_account__parent_toptier_agency__toptier_code": self.code})
+
+    def clone(self, code, positive, positive_naics, negative_naics):
+        return TASNode(code, positive, positive_naics, negative_naics)
+
+
+class TreasuryAccounts:
+    underscore_name = "treasury_account_components"
+
+    @staticmethod
+    def build_tas_codes_filter(queryset, tas_filters):
+        """
+        Filter the queryset to take advantage of the GIN indexed integer array of
+        treasury_appropriation_account treasury_account_identifiers by:
+
+            - Grabbing the list of all treasury_account_identifiers that match the filters provided.
+            - Assembling an integer array overlap (&&) query to search for those integers.
+        """
+
+        tas_qs = Q()
+        for tas_filter in tas_filters:
+            tas_qs |= Q(**{TAS_COMPONENT_TO_FIELD_MAPPING[k]: v for k, v in tas_filter.items()})
+
+        # No sense continuing if this resulted in an empty Q statement for whatever reason.
+        if not tas_qs:
+            return queryset
```

Comment:
This seems like a problem.  If we don't build a Q we return the queryset.  Should it return an empty Q instead?

Reply:
@BrianZito , this is still an issue.  Trying to filter a queryset with another queryset throws an exception.  I think this needs to return an empty Q (probably just `tas_qs`).

Reply:
I'm pretty sure (and will unit test) that this can be removed altogether. If the user provides an empty list of Treasury Accounts, they should NEVER get results (unless they are also using tas_codes), which the rest of this function will cover.

Reply:
This is true.  I couldn't find a code path that got me to this line so I had to hack it, however, I don't think it hurts anything to program a little defensively and just return an empty Q.  There may be no code path today, but who knows what crazy scheme tomorrow's Brian will cook up.

Reply:
Thanks.  I think your solution is more correct.  👍